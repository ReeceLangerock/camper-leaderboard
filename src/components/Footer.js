import React from "react";
import styled from "styled-components";

const FooterStyled = styled.div`
  background-color: #f4faff;
  color: #535657;
  font-size: 1.4rem;
  align-items: center;
  padding: 12px;
  align-content: center;
  text-align: center;
  vertical-align: middle;

  a {
    color: #3bb273;
    text-decoration: none;
  }
`;

const I = styled.i`
  margin-left: 10px;

  color: black;
  cursor: pointer;
`;

export class Footer extends React.Component {
  handleIconClick(link) {
    window.open("https://github.com/ReeceLangerock/camper-leaderboard", "_blank");
  }
  render() {
    const currentYear = new Date().getFullYear();

    return (
      <FooterStyled className="Footer">
        Â© {currentYear}{" "}
        <a href="http://reecelangerock.com" rel="noopener noreferrer" target="_blank">
          Reece Langerock
        </a>
        <I className={`fa fa-github`} aria-hidden="false" onClick={() => this.handleIconClick()} />
      </FooterStyled>
    );
  }
}
export default Footer;
