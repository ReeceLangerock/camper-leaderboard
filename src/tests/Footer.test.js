import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";

import Footer from "./../components/Footer";
import enzymeSerializer from "enzyme-to-json/serializer";

expect.addSnapshotSerializer(enzymeSerializer);

describe("Footer", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer />, div);
  });

  let component;
  beforeEach(() => {
    const props = {};
    component = mount(<Footer {...props} />);
  });

  it("has the correct class", () => {
    expect(component.hasClass("Footer")).toEqual(true);
  });
});

describe("Snapshot Footer", () => {
  it("matches snapshot", () => {
    const comp = mount(<Footer />);
    expect(comp).toMatchSnapshot();
  });
});
