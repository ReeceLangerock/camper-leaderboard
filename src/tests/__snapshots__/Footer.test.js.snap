// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Footer matches snapshot 1`] = `
<Footer>
  <div
    className="Footer"
  >
    Â© 
    2017
     
    <a
      href="http://reecelangerock.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Reece Langerock
    </a>
  </div>
</Footer>
`;
